client/src/components/categorical/category.js

```diff
@@ -159,22 +159,16 @@ class Category extends React.Component {
     or return an error type.
     */
     let error = false;
-
-    if (!name) {
-      error = false;
-    }
-
-    const { metadataField, universe } = this.props;
-    const { obsByName } = universe.schema.annotations;
-
-    if (obsByName[metadataField].categories.indexOf(name) !== -1) {
-      error = "duplicate";
-    }
-
-    if (!AnnotationsHelpers.isLegalAnnotationName(name)) {
-      error = "characters";
+    if (name) {
```

Comment:
This probably could be moved outside the function itself, depending on how many times this function is called. Imagining: 

`name && errorCheck()`

Reply:
That doesn't add much to the verbosity of the inline check

Reply:
it all depends on what you want the default behavior to be.  If you look at line 510, you see that we occasionally say `!name && errorCheck()`, so you have to pick a default.  In this case, most of the invocations preferred that and error only occurs when name is a non-zero length string.

I guess we could also pass in the desired behavior as an arg, eg, `errorCheck(val, nullIsAnError=true)`

I'm not dogmatic on this - fix to match your own taste...